doc/programming/language/statements.rst

```diff
@@ -41,6 +41,19 @@ will be carried along with the exception.
 
 .. include:: /autogen/types/generic-begin.rst
 
+.. _statement_block:
+
+``block``
+---------
```

Comment:
This seem misleading as `block` is not a keyword. Two ideas:

1. Say `BLOCK` instead, to match how we refer to it elsewhere.
2. Add an explanation at the beginning of the section inside the preface (between 5.2.6 and 5.2.6.1 here: https://docs.zeek.org/projects/spicy/en/latest/programming/language/statements.html)
3. Given that this is primarily about variable scope, add it to https://docs.zeek.org/projects/spicy/en/latest/programming/language/variables.html

I think I'd go for (3).

Reply:
I agree, 3 seems nicer. I changed it to go there. Feel free to wordsmith it as much as you want